libtransmission/tr-dht.cc

```diff
@@ -216,18 +381,20 @@ static void dht_bootstrap(tr_session* session, std::vector<uint8_t> nodes, std::
             nap(15);
         }
 
-        if (bootstrap_done(my_session, 0))
+        if (isBootstrapDone(session, 0))
         {
             break;
         }
     }
+    TR_ASSERT(walk4 = std::data(nodes4) + std::size(nodes4));
```

Comment:
@ckerr Is it supposed to be an equality check, not assignment?
It crashes for me when empty `nodes4`/`nodes6` are passed.
`std::data` returns `nullptr`, this whole expr is evaluated to `nullptr + 0` -> `nullptr` -> `0` -> `false` and `TR_ASSERT` fails.

Reply:
Fixed by @Coeur in f27e1b52f46c0ea207a195af6ee1940bc1bd8ed8